akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/Effect.scala

```diff
@@ -230,6 +233,64 @@ object Effect {
 
   final case class TimerCancelled(key: Any) extends Effect
 
+  /**
+   * The behavior initiated an ask of a targeted recipient.
+   */
+  final class AskInitiated[Req, Res, T](
+      val target: RecipientRef[Req],
+      val responseClass: Class[Res],
+      createMessage: ActorRef[Res] => Req,
+      mapResponse: Try[Res] => T)
+      extends Effect
+      with Product2[RecipientRef[Req], Class[Res]]
```

Comment:
Make it a case class like the other effects and it will get Product, equals etc. the things that should not be part of object equality can be in a separate parameter list to the constructor just like how it is done with `TimerScheduled`